mungegithub/mungers/release-note-label.go

```diff
@@ -40,7 +40,8 @@ const (
 	releaseNoteActionRequired = "release-note-action-required"
 	releaseNoteExperimental   = "release-note-experimental"
 
-	releaseNoteFormat = `Adding ` + doNotMergeLabel + ` because the release note process has not been followed.
+	releaseNoteMissingLabel = doNotMergePrefix + "/release-note-label-missing"
```

Comment:
What does that goal mean in terms of these changes?

Reply:
a) Make sure the submit queue will not merge anything with a release-note-needed label (maybe rename it do-not-merge/release-note-needed
b) Remove the logic to apply the do-not-merge label when a PR gets approved with release-note-needed.

It is too punitive because there is no way for the author/reviewers/approvers to remove the label unless they have write access to the repo.

Reply:
OK, I will do that in #4196